File: dsl-api/src/main/java/cbs/dsl/api/context/TransactionContext.java
package cbs.dsl.api.context;

import lombok.Builder;

import java.util.Map;

@Builder(toBuilder = true)
public record TransactionContext(
    String eventNumber,
    String performedBy,
    Map<String, Object> params,
    HelperEvaluator helperEvaluator,
    TransactionEvaluator transactionEvaluator) {

  public TransactionContext put(String key, Object value) {
    params.put(key, value);
    return this;
  }

  public Object get(String key) {
    return params.get(key);
  }

  public TransactionContext copy() {
    return toBuilder().build();
  }

  public Object runTransaction(String code, Map<String, Object> values) {
    return transactionEvaluator.evaluate(code, values);
  }
}

File: dsl-api/src/main/kotlin/cbs/dsl/api/context/TransactionContext.kt
package cbs.dsl.api.context

open class TransactionContext(
  eventCode: String,
  workflowExecutionId: Long,
  performedBy: String,
  dslVersion: String,
  eventParameters: Map<String, Any>,
  val isResumed: Boolean,
) : EnrichmentContext(eventCode, workflowExecutionId, performedBy, dslVersion, eventParameters) {

  open fun delegate(): Unit = Unit
}
